backend/api/job.proto

```diff
@@ -130,10 +155,31 @@ message ListJobsRequest {
   // Ascending by default.
   string sort_by = 3;
 
-  // What resource reference to filter on.
-  // E.g. If listing job for an experiment, the query string would be
-  // resource_reference_key.type=EXPERIMENT&resource_reference_key.id=123
-  ResourceKey resource_reference_key = 4;
+  // Optional input. The namespace the jobs belong to. 
+  string namespace = 4;
+
+  // A url-encoded, JSON-serialized Filter protocol buffer (see
+  // [filter.proto](https://github.com/kubeflow/pipelines/blob/master/backend/api/filter.proto)).
+  string filter = 5;
+}
+
+message ListJobsByExperimentRequest {
```

Comment:
The only difference between `ListJobsByExperimentRequest` and `ListJobsRequest` is the one field (experiment_id vs namespace). Would it make sense to reuse the rest?

Reply:
I still prefer defining them separately, in case either of them will be updated in the future, there won't be dependency issue.